src/cli/subcommands/sync_cmd.rs

```diff
@@ -185,8 +113,85 @@ impl SyncCommands {
     }
 }
 
-fn format_tipset_cids(cids: &str) -> &str {
-    if cids.is_empty() { "[]" } else { cids }
+/// Prints the sync status report details.
+/// `line_count` is mutable and incremented for each line printed, used for clearing in `Wait`.
+/// Pass `&mut 0` if line counting/clearing is not needed (like in `Status` or final print).
+fn print_sync_report_details(
+    report: &SyncStatusReport,
+    line_count: &mut usize,
+) -> anyhow::Result<()> {
+    println!(
+        "Status: {:?} ({} epochs behind)",
+        report.get_status(),
+        report.get_epochs_behind()
+    );
+    *line_count += 1;
+
+    let head_key_str = report
+        .get_current_chain_head_key()
+        .map(tipset_key_to_string)
+        .unwrap_or_else(|| "[unknown]".to_string());
+    println!(
+        "Node Head: Epoch {} ({})",
+        report.get_current_chain_head_epoch(),
+        head_key_str
+    );
+    *line_count += 1;
+
+    println!("Network Head: Epoch {}", report.get_network_head_epoch());
+    *line_count += 1;
+
+    println!("Last Update: {}", report.get_last_updated().to_rfc3339());
+    *line_count += 1;
+
+    // Print active sync tasks (forks)
+    let active_forks = report.get_active_forks();
+    if active_forks.is_empty() {
+        println!("Active Sync Tasks: None");
+        *line_count += 1;
+    } else {
+        println!("Active Sync Tasks:");
+        *line_count += 1;
+        let mut sorted_forks = active_forks.clone();
+        sorted_forks.sort_by_key(|f| std::cmp::Reverse(f.target_epoch));
+        for fork in &sorted_forks {
+            // Assuming print_fork_sync_info exists and increments line_count internally if needed
+            // If print_fork_sync_info doesn't increment, adjust line_count here.
+            // For simplicity, assuming it behaves as needed or is adjusted elsewhere.
+            print_fork_sync_info(fork, line_count)?;
+        }
+    }
+    Ok(())
+}
+
+fn print_fork_sync_info(fork: &ForkSyncInfo, line_count: &mut usize) -> anyhow::Result<()> {
+    let total_epochs_for_this_fork = fork
+        .target_epoch
+        .saturating_sub(fork.target_sync_epoch_start);
+    println!(
+        "  - Fork Target: {} ({}), Stage: {}, Syncing Range: [{}..{}] ({} epochs)",
+        fork.target_epoch,
+        tipset_key_to_string(&fork.target_tipset_key),
+        &fork.stage,
+        fork.target_sync_epoch_start,
+        fork.target_epoch,
+        total_epochs_for_this_fork
+    );
+    if *line_count > 0 {
+        // Only increment if we are in the Wait command context
+        *line_count += 1;
```

Comment:
from this it's not clear how we determine we are in `Wait` context, adding a comment above conditional check will make more sense

Reply:
also using `Option::None` instead of 0 is better

Reply:
Removed this part.